contracts/CodexRecordMetadata.sol

```diff
@@ -170,4 +170,19 @@ contract CodexRecordMetadata is ERC721Token {
 
     return bstr;
   }
+
+  /**
+   * @dev Returns whether or not a bytes32 array is empty (all null-bytes)
+   * @param _data bytes32 The array to check
+   * @return bool Whether or not the array is empty
+   */
+  function bytes32IsEmpty(bytes32 _data) internal pure returns (bool) {
+    for (uint256 i = 0; i < 32; i++) {
```

Comment:
You might save gas if you make this a `uint8`? Not sure if memory allocation affects gas used.

Reply:
Outside of structs all uint types take up the same amount of storage as uint256, so nothing saved by doing so